src/test/spec/unified_runner/entity.rs

```diff
@@ -267,7 +284,13 @@ impl Deref for ClientEntity {
     type Target = Client;
 
     fn deref(&self) -> &Self::Target {
-        &self.client
+        match &self.client {
+            ClientEntityState::Client(c) => c,
+            ClientEntityState::Dropped => panic!(
```

Comment:
it didn't seem great to panic here (and the docs say implementations of this method should never fail), but this can only happen in a very specific case where a test tries to use a client that was closed via a `close` operation, which is documented as something that tests should not do in my spec PR. so it seemed worth the ergonomic tradeoff of implementing deref to leave this.